snowpack/src/util.ts

```diff
@@ -302,17 +302,19 @@ export function findMatchingAliasEntry(
   }
 }
 
-/** Get full extensions of files */
-export function getExt(fileName: string) {
-  return {
-    /** base extension (e.g. `.js`) */
-    baseExt: path.extname(fileName).toLocaleLowerCase(),
-    /** full extension, if applicable (e.g. `.proxy.js`) */
-    expandedExt: path.basename(fileName).replace(/[^.]+/, '').toLocaleLowerCase(),
-  };
+export function getExtensionMatch(fileName: string, extensionMap: Record<string, string>): string | undefined {
+  let extensionMatch;
+  let extensionMatchIndex = 0;
+  while (!extensionMatch && extensionMatchIndex > -1) {
```

Comment:
If I’m reading this right, will this loop break on the first failed lookup of a file extension (`extensionMatchIndex = fileName.indexOf` below)?

Reply:
The loop breaks if either:
- `!extensionMatch` - a match is found 
- `extensionMatchIndex  > -1` - `fileName.indexOf` fails to find a match

Reply:
Oh no you‘re stepping through extensions and getting the largest match you can. This looks fine; it’s just hard to read (would a `.split()` be easier?)

Reply:
Will add better docs!